.github/workflows/rust.yml

```diff
@@ -0,0 +1,86 @@
+name: Rust
+
+on:
+  push:
+    branches:
+    - '*'
+    tags-ignore:
+    - 'v*'
+  pull_request:
+    branches:
+    - '*'
```

Comment:
It's common to have either `on: push` or `on: pull_request`. Having both causes very duplicate CI runs most of the time. We're usually using `push`.

Reply:
Yeah I was just copying our innernet workflow, but my preferred setup is like so:

```yaml
on:
  push:
    branches:
      - main
  pull_request:
```

I'll change it to that.